Task 2 5 input validation

Doenator310 opened the pull request:
A js file was created to check the password, username and email address, 
that means Task 2.5 is finished

Commit:
Input Validierung implementiert?!

Commit:
Renamed inputvalidator.html to inputvalidator_example.html because it is an example

Commit:
Summary:
+Added inputvalidator.js as tool to verify the correct length and format of a password, username and email, on the server and on the web browser.
+Added inputvalidator_example.html to test the functionality server sided
+api.js handles an validateInput request, for server side validation

Doenator310 committed:
Merge branch 'master' into task_2_5_input_validation

Commit:
+Added link to source for password regex
+Formatting has been revised

Commit:
remove accidently added dependency

Commit:
Merge branch 'task_2_5_input_validation' of https://github.com/DavidRisch/kicker into task_2_5_input_validation

Doenator310 committed:
Merge branch 'master' into task_2_5_input_validation

DavidRisch commented on app.js:
```diff
@@ -1,13 +1,13 @@
 const dotenv = require('dotenv')
-const path = require('path')
+const path   = require('path')
```
Incorrect formatting, see CONTRIBUTING.md

DavidRisch commented on package.json:
```diff
@@ -9,6 +9,8 @@
-    "mysql": "^2.18.1"
+    "jquery": "^3.5.0",
+    "mysql": "^2.18.1",
+    "standard": "^14.3.3"
```
standard is no dependency, it is not required to execute this project. Theoretically it could be added as `devDependencies`. But unless you can name an advantage this would bring, please remove the dependency completely.

DavidRisch commented on page/page.js:
```diff
@@ -4,6 +4,11 @@ function init (app) {
     require('./example').page(req, res)
   })
+
+  app.get('/inputvalidator', function (req, res) {
```
Please change the name to `input_validator`. This applies to all files/names in this pr

DavidRisch commented:
Please add the email you use in your commits `david@dieterhenkel.de`to github.
https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/adding-an-email-address-to-your-github-account

Doenator310 committed:
-Removed "standard" npm dependency [that wasnt added intentionally]
-fixed formatting errors
+renamed files

Doenator310 committed:
Merge branch 'master' into task_2_5_input_validation

Doenator310 committed:
Merge "newer" branch 'task_2_5_input_validation' of https://github.com/DavidRisch/kicker into task_2_5_input_validation

Doenator310 committed:
-removed debugging info

Doenator310 committed:
+Formatting

Doenator310 commented:
Added the correct email to my local git configuration.

DavidRisch commented on api/api.js:
```diff
@@ -17,6 +17,20 @@ function init (app) {
+        response = {
+          'valid-email': validator.isValidEmail(email),
+          'valid-name': validator.isValidUserName(name),
+          'secure-password': validator.isSecurePassword(password)
```
Please keep your naming style: `valid-password`

DavidRisch commented on js/input_validator.js:
```diff
@@ -0,0 +1,20 @@
+}
+
+function isValidUserName (userNameText) {
+//  A-Z, a-z, 0-9, _ - ist erlaubt, mindestens 5 Zeichen
```
Please write all comments in English

DavidRisch commented on js/input_validator.js:
```diff
@@ -0,0 +1,20 @@
+
+function isSecurePassword (passwordText) {
+//  Länge mindestens 8
+//  Mind. 1 Sonderzeichen, 1 Großbuchstabe und 1 Zahl
```
Please write all comments in English

DavidRisch commented on package.json:
```diff
@@ -9,6 +9,7 @@
     "deasync": "^0.1.19",
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
+    "jquery": "^3.5.0",
```
As far as I can tell this dependency is never used. See https://stackoverflow.com/a/29537014. Alternatively you can remove the dependency.

DavidRisch commented on js/input_validator.js:
```diff
@@ -0,0 +1,20 @@
+//  Länge mindestens 8
+//  Mind. 1 Sonderzeichen, 1 Großbuchstabe und 1 Zahl
+//  A Modified version of https://stackoverflow.com/a/5142164
+  const passwordRegex = new RegExp(/^(?=.*[A-Z])(?=.*[!@#$&*])(?=.*[0-9])(?=.*[a-z]).{8,}$/)
```
Why are special symbols restricted to `!@#$&*`? Suggestion: `[^A-Za-z0-9]`
Please document the exact rules in a comment.

DavidRisch commented on js/input_validator.js:
```diff
@@ -0,0 +1,20 @@
+function isValidEmail (emailString) {
+//  https://www.ietf.org/rfc/rfc5322.txt
```
This source does not back up the claimed regex (please provide page number if I am wrong). Please provide the actual source. https://emailregex.com/ ?

DavidRisch commented on html/input_validator_example.html:
```diff
@@ -0,0 +1,38 @@
+    <meta charset="UTF-8">
+    <title>Example</title>
+    <link type="text/css" rel="stylesheet" href="css/style.css"/>
+    <script type="text/javascript" src="js/input_validator.js"></script>
```
As far as I can tell this has no effect. The same is true for `jquery.validate.min.js`. You need to actually use them

DavidRisch reviewed (changes requested):
See requested changes

Doenator310 committed:
-Removed unused jquery dependencys from package.json and input_validator_example.html
+Changed name in api.js as requested
+Added more Comments to the Regex functions
+Added Regex source

DavidRisch commented:
From CONTRIBUTING.md:
Git commits must start with a capital letter [...]. [Helpful guide](https://chris.beams.io/posts/git-commit/)

DavidRisch reviewed (approved):
looks good
